ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java

```diff
@@ -33,462 +33,99 @@
-     * values. This notification have got an unread state, a info type. This notification does not delegate any actions in response to
-     * opening and closing of a notification.
+     * Get notification event listener.
      *
-     * @param message
-     *         notification's message
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
+     * @return notification event listener
      */
-    public Notification(@NotNull String message, @NotNull Status status, boolean important) {
-        this(message, status, important, null, null);
+    public NotificationListener getListener() {
+        return listener;
     }
 
-    /**
-     * Create notification with message, status, note about important this one and action delegate on opening of a notification.
-     * Other parameters will be added with default values. This notification have got an unread state, a info type. This notification
-     * delegates some actions in response to opening of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
-     * @param openHandler
-     *         delegate that provides some actions when opening notification
-     */
-    public Notification(@NotNull String message, @NotNull Status status, boolean important, @NotNull OpenNotificationHandler openHandler) {
-        this(message, INFO, status, UNREAD, new Date(), important, openHandler, null);
+    public void setListener(NotificationListener listener) {
+        if (listener != null) {
+            this.listener = listener;
+            setUnreadState();
+        }
     }
 
-    /**
-     * Create notification with message, status, note about important this one and action delegate on closing of a notification.
-     * Other parameters will be added with default values. This notification have got an unread state, a info type. This notification
-     * delegates some actions in response to closing of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Status status, boolean important,
-                        @NotNull CloseNotificationHandler closeHandler) {
-        this(message, INFO, status, UNREAD, new Date(), important, null, closeHandler);
+    protected void setUnreadState() {
+        //lets collect all calls instead of continuously calling update observers
+        setUnreadStateTask.delay(200);
```

Comment:
Why its 200 ?
It seems that it should be moved to kind of _constants_ place and explained